src/main/java/com/iexec/common/docker/DockerRunRequest.java

```diff
@@ -72,39 +74,48 @@ public String getStringArgsCmd() {
         return devices != null ? new ArrayList<>(devices) : List.of();
     }
 
-    // override builder's isSgx() & devices() methods
+    public SgxDriverMode getSgxDriverMode() {
+        return sgxDriverMode != null ? sgxDriverMode : SgxDriverMode.NONE;
+    }
+
+    // override builder's sgxDriverMode() & devices() methods
     public static class DockerRunRequestBuilder { 
-        private boolean isSgx;
+        private SgxDriverMode sgxDriverMode;
         private List<Device> devices;
 
         /**
-         * Add an SGX device when isSgx is true.
+         * Depending on SGX driver mode, do the following:
+         * <ul>
+         *     <li>If mode is {@literal null} or {@link SgxDriverMode#NONE},
+         *     simply stores the mode {@link SgxDriverMode#NONE};</li>
+         *     <li>If mode is {@link SgxDriverMode#LEGACY} or {@link SgxDriverMode#NATIVE},
+         *     stores this mode and stores a list of devices defined in the related enum value.</li>
+         * </ul>
          * 
-         * @param isSgx
-         * @return
+         * @param sgxDriverMode SGX driver mode
+         * @return This {@link DockerRunRequestBuilder} with updated fields.
          */
-        public DockerRunRequestBuilder isSgx(boolean isSgx) {
-            this.isSgx = isSgx;
-            if (!isSgx) {
+        public DockerRunRequestBuilder sgxDriverMode(SgxDriverMode sgxDriverMode) {
+            this.sgxDriverMode = Objects.requireNonNullElse(sgxDriverMode, SgxDriverMode.NONE);
+            if (!SgxDriverMode.isDriverModeNotNone(sgxDriverMode)) {
                 return this;
             }
             if (this.devices == null) {
                 this.devices = new ArrayList<>();
             }
-            Device sgxDevice = new Device(
-                    SgxUtils.SGX_CGROUP_PERMISSIONS,
-                    SgxUtils.SGX_DEVICE_PATH,
-                    SgxUtils.SGX_DEVICE_PATH);
-            this.devices.add(sgxDevice);
+
+            for (String deviceName : sgxDriverMode.getDevices()) {
+                this.devices.add(Device.parse("/dev/" + deviceName));
```

Comment:
Do we need to manage permissions here?

Reply:
I don't think so. Old way was to declare them explicitly, new way is to let `Device.parse` manage them - which currently is the same as before (`rwm`).